src/Form/Form.js

```diff
@@ -786,7 +780,17 @@ class Form {
             }
 
             if (autofillData) {
-                this.autofillInput(input, autofillData, dataType)
+                const variant = getInputVariant(input)
+                if (!variant) {
+                    return this.autofillInput(input, autofillData, dataType)
+                }
+
+                if (variant === 'new' && AUTOGENERATED_KEY in data) {
+                    return this.autofillInput(input, autofillData, dataType)
+                }
+                if (variant === 'current' && !(AUTOGENERATED_KEY in data)) {
+                    return this.autofillInput(input, autofillData, dataType)
+                }
```

Comment:
Fill all related fields, so if there's a repeat new password field we fill it in one go.